src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java

```diff
@@ -77,53 +516,1905 @@ static void dropAll() throws SQLServerException, SQLException {
         }
     }
 
+	private void populateBinaryNormalCase(LinkedList<byte[]> byteValues) throws SQLException {
+		String sql = "insert into " + binaryTable + " values( " + "?,?,?," + "?,?,?," + "?,?,?," + "?,?,?," + "?,?,?"
+				+ ")";
+
+		pstmt = (SQLServerPreparedStatement) Util.getPreparedStmt(con, sql, stmtColEncSetting);
+
+		// binary20
+		for (int i = 1; i <= 3; i++) {
+			if (null == byteValues) {
+				pstmt.setBytes(i, null);
+			} else {
+				pstmt.setBytes(i, byteValues.get(0));
+			}
+		}
+
+		// varbinary50
+		for (int i = 4; i <= 6; i++) {
+			if (null == byteValues) {
+				pstmt.setBytes(i, null);
+			} else {
+				pstmt.setBytes(i, byteValues.get(1));
+			}
+		}
+
+		// varbinary(max)
+		for (int i = 7; i <= 9; i++) {
+			if (null == byteValues) {
+				pstmt.setBytes(i, null);
+			} else {
+				pstmt.setBytes(i, byteValues.get(2));
+			}
+		}
+
+		// binary(512)
+		for (int i = 10; i <= 12; i++) {
+			if (null == byteValues) {
+				pstmt.setBytes(i, null);
+			} else {
+				pstmt.setBytes(i, byteValues.get(3));
+			}
+		}
+
+		// varbinary(8000)
+		for (int i = 13; i <= 15; i++) {
+			if (null == byteValues) {
+				pstmt.setBytes(i, null);
+			} else {
+				pstmt.setBytes(i, byteValues.get(4));
+			}
+		}
+
+		pstmt.execute();
+	}
+
+	private void populateBinarySetObject(LinkedList<byte[]> byteValues) throws SQLException {
+		String sql = "insert into " + binaryTable + " values( " + "?,?,?," + "?,?,?," + "?,?,?," + "?,?,?," + "?,?,?"
+				+ ")";
+
+		pstmt = (SQLServerPreparedStatement) Util.getPreparedStmt(con, sql, stmtColEncSetting);
+
+		// binary(20)
+		for (int i = 1; i <= 3; i++) {
+			if (null == byteValues) {
+				pstmt.setObject(i, null, java.sql.Types.BINARY);
+			} else {
+				pstmt.setObject(i, byteValues.get(0));
+			}
+		}
+
+		// varbinary(50)
+		for (int i = 4; i <= 6; i++) {
+			if (null == byteValues) {
+				pstmt.setObject(i, null, java.sql.Types.BINARY);
+			} else {
+				pstmt.setObject(i, byteValues.get(1));
+			}
+		}
+
+		// varbinary(max)
+		for (int i = 7; i <= 9; i++) {
+			if (null == byteValues) {
+				pstmt.setObject(i, null, java.sql.Types.BINARY);
+			} else {
+				pstmt.setObject(i, byteValues.get(2));
+			}
+		}
+
+		// binary(512)
+		for (int i = 10; i <= 12; i++) {
+			if (null == byteValues) {
+				pstmt.setObject(i, null, java.sql.Types.BINARY);
+			} else {
+				pstmt.setObject(i, byteValues.get(3));
+			}
+		}
+
+		// varbinary(8000)
+		for (int i = 13; i <= 15; i++) {
+			if (null == byteValues) {
+				pstmt.setObject(i, null, java.sql.Types.BINARY);
+			} else {
+				pstmt.setObject(i, byteValues.get(4));
+			}
+		}
+
+		pstmt.execute();
```

Comment:
I created Util.close(ResultSet, Statement, Connection) that employs good practice for closing those 3 objects, and applied it to the test cases.